pallets/parachain-staking/src/lib.rs

```diff
@@ -980,29 +1005,43 @@ pub mod pallet {
 						// since it is assumed that they were removed incrementally before only the
 						// last delegation was left.
 						<DelegatorState<T>>::remove(&bond.owner);
+						T::Currency::remove_lock(DELEGATOR_LOCK_ID, &bond.owner);
 					} else {
 						<DelegatorState<T>>::insert(&bond.owner, delegator);
+						// update locked balance to match adjusted total staked
+						T::Currency::set_lock(
+							DELEGATOR_LOCK_ID,
+							&bond.owner,
+							remaining,
+							WithdrawReasons::all(),
+						);
 					}
+				} else {
+					// TODO: review. we assume here that this delegator has no remaining staked
+					// balance, so we ensure the lock is cleared
+					T::Currency::remove_lock(DELEGATOR_LOCK_ID, &bond.owner);
 				}
```

Comment:
It's not clear what should happen here. `remove_lock` is probably wrong, though. When `rm_delegation` is called, it will return `None` iff the matching delegation can't be found, but this should never be the case: it would indicate a logic error somewhere else resulting in inconsistent state. Is that right, @4meta5?

Assuming that's the case, I suppose we ignore this codepath as we did previously.